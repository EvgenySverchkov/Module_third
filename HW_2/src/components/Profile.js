import React from 'react';
import "./Profile.css";

export default class Profile extends React.Component{
	render(){
		return (<>
			<form className="form-inline" onSubmit={(e)=>{
				e.preventDefault();
				this.props.header(e.target.elements.userId.value);
			}}>
			<div className="form-group mb-2">
				<label forhtml='userId' className="col-lm-2 col-form-label">Input userId</label>
			</div>
			<div className="form-group mx-sm-3 mb-2">
				<input type="number" id="userId" className="form-control" placeholder="Eneter id"></input>
			</div>
			
			<button type="submit" className="btn btn-primary mb-2">Submit</button>
			</form>
		</>);
	}
}