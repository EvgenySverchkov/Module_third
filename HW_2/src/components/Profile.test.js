import React from 'react';
import { shallow } from 'enzyme';

import Profile from './Profile.js';

describe("Profile tets", ()=>{
	it("snapshot test for Profile", ()=>{
	const wraper = shallow(<Profile />);
	expect(wraper.debug()).toMatchSnapshot();
});

	it("shoud call form handler", ()=>{
		const mock = jest.fn();
		const props = {header: mock};
		const wraper = shallow(<Profile {...props}/>);

		const form = wraper.find('.form-inline');

		form.simulate('submit', {
			preventDefault: ()=>{},
			target: {
				elements:{
					userId:{
						value:0
					}
				}
			}
		});
		expect(mock).toHaveBeenCalledTimes(1);
	});
});