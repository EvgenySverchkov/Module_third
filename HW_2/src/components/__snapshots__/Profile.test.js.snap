// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile tets snapshot test for Profile 1`] = `
"<Fragment>
  <form className=\\"form-inline\\" onSubmit={[Function: onSubmit]}>
    <div className=\\"form-group mb-2\\">
      <label forhtml=\\"userId\\" className=\\"col-lm-2 col-form-label\\">
        Input userId
      </label>
    </div>
    <div className=\\"form-group mx-sm-3 mb-2\\">
      <input type=\\"number\\" id=\\"userId\\" className=\\"form-control\\" placeholder=\\"Eneter id\\" />
    </div>
    <button type=\\"submit\\" className=\\"btn btn-primary mb-2\\">
      Submit
    </button>
  </form>
</Fragment>"
`;
